templates/facetDemoScript.template.hbs

```diff
@@ -0,0 +1,165 @@
+import { {{titleCase name}}Facet, ILiFi } from '../../typechain'
+import {
+  getEnvVar,
+  normalizePrivateKey
+} from './utils/demoScriptHelpers'
+import { getRpcUrl, getViemChain, SupportedChain } from './utils/demoScriptChainConfig'
+
+// #region ABIs
+
+const ERC20_ABI = erc20Artifact.abi as Narrow<typeof erc20Artifact.abi>
+const {{constantCase name}}_FACET_ABI = {{camelCase name}}FacetArtifact.abi as Narrow<typeof {{camelCase name}}FacetArtifact.abi>
+const EXAMPLE_ABI = [{...}] // optional define example ABI
+
+// #endregion
+
+dotenv.config()
+
+async function main() {
+  // === Set up parameters, retrieve env vars and set up clients ===
+  // `srcChain` is the source chain for the operation (change as needed).
+  // RPC URL and private key are dynamically fetched based on the chain.
+  const srcChain: SupportedChain = "mainnet"; // Set source chain
+  const RPC_URL = getRpcUrl(srcChain)
+  const PRIVATE_KEY = getEnvVar('PRIVATE_KEY')
+  const typedPrivateKey = normalizePrivateKey(PRIVATE_KEY)
+
+  const publicClient = createPublicClient({
+    chain: getViemChain(srcChain),
+    transport: http(RPC_URL),
+  })
+  const walletAccount = privateKeyToAccount(typedPrivateKey)
+  const walletClient = createWalletClient({
+    chain: getViemChain(srcChain),
+    transport: http(RPC_URL),
+    account: walletAccount,
+  })
+  const signerAddress = walletAccount.address
+
+  // === Contract addresses ===
+  // LIFI_ADDRESS is the address of the LiFi Diamond contract for bridging.
+  // SRC_TOKEN_ADDRESS is the token to be bridged (set its address for the source chain).
+  // EXAMPLE_ADDRESS is a custom contract address for a contract that your facet interacts with.
+  const LIFI_ADDRESS = deployments.LiFiDiamond as `0x${string}` // Set correct LiFiDiamond address
+  const SRC_TOKEN_ADDRESS = '' as `0x${string}` // Set the source token address here.
+  const EXAMPLE_ADDRESS = config.mainnet.example as `0x${string}` // Update for the correct chain.
+
+  // === Instantiate contracts ===
+  const srcTokenContract = getContract({
+    address: SRC_TOKEN_ADDRESS,
+    abi: ERC20_ABI,
+    client: { public: publicClient, wallet: walletClient }
+  })
+
+  const exampleContract = getContract({
+    address: EXAMPLE_ADDRESS,
+    abi: EXAMPLE_ABI,
+    client: { public: publicClient, wallet: walletClient }
+  })
+
+  const lifiDiamondContract = getContract({
+    address: LIFI_ADDRESS,
+    abi: {{constantCase name}}_FACET_ABI,
+    client: { public: publicClient, wallet: walletClient }
+  })
+
+  const srcTokenName = await srcTokenContract.read.name() as string
+  const srcTokenSymbol = await srcTokenContract.read.symbol() as string
+  const amount = parseUnits('10', 18) // 10 * 1e18
+
+  console.info(`\Bridge ${amount} ${srcTokenName} (${srcTokenSymbol}) from ${srcChain} --> {DESTINATION CHAIN NAME}`) // Set destination chain name
+  console.info(`Connected wallet address: ${signerAddress}`)
+
+  // === Check token balance ===
```

Comment:
balance and allowance checks and error handling could also be moved to a helper function as this is probably gonna be used in every script